src/box/iproto.cc

```diff
@@ -1714,6 +1801,7 @@ static void
 tx_reply_iproto_error(struct cmsg *m)
 {
 	struct iproto_msg *msg = tx_accept_msg(m);
+	assert(in_txn() == NULL);
```

Comment:
This can happen if you send a bad message with stream_id the same as an existing valid stream having a transaction active. The error iproto_msg will be serialized into the stream like all the others, will reach this place, and get a transaction in `tx_accept_msg()`. I suggest not to change the flow of this request and simply add `tx_end_msg()` in the end of it. Let the transaction pass through it. You can also try to add a test on it.

Reply:
Yes fixed